src/ui/public/visualize/visualization.js

```diff
@@ -23,16 +23,23 @@ uiModules
         vis: '=',
         visData: '=',
         uiState: '=?',
+        // If set to true (default) the visualization directive will listen for
+        // several changes in the data and uiState to trigger a render. If set to
+        // false (boolean value, not a falsy value), this directive won't listen
+        // for any changes and require to be notified by the 'render' event broadcasted
+        // to its scope, that it needs to rerender this. Usually when using this as
+        // a consumer you don't want to change the default behavior.
+        listenOnChange: '<',
         searchSource: '='
       },
       template: visualizationTemplate,
       link: function ($scope, $el) {
-        const resizeChecker = new ResizeChecker($el);
```

Comment:
Earlier we where checking for resize events of the `visualization` element, but only if we were not running inside `visualize`. We now properly check on the actual div element, the chart is rendered to. That way we also respect, that there are might be other elements (e.g. the legend) inside this directive, that might take away space from the chart.

This did work by chance, since we stored the state of the legend (open/closed) inside the uiState and thus trigger a render anyway, when it changed. If we wouldn't have stored the legend state in the `uiState` `visualization` would have never detected, that the chart needs a redraw, because we were looking on the container's size (which wouldn't change if the legend opens/closes) and not on the actual chart size (which does change in that case).